.github/workflows/ci.yml

```
@@ -128,10 +127,28 @@ jobs:
   performance-tests:
     name: Performance Tests
     runs-on: ubuntu-22.04
+    permissions:
+      contents: write
     steps:
       - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
       - uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
         with:
           go-version-file: 'go.mod'
       - name: Run Performance Tests
-        run: make performance-test
+        run: |
+          make performance-test
+      - name: Store benchmark result
+        uses: benchmark-action/github-action-benchmark@v1
```

As far as I understand it's best practice to pin actions to a full length SHA

```suggestion
        uses: benchmark-action/github-action-benchmark@cc9ac13ce81036c9b67fcfe2cb95ca366684b9ea # v1.19.3
```

Ref: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions